x/concentrated-liquidity/keeper_test.go

```diff
@@ -154,6 +156,116 @@ func (s *KeeperTestSuite) initializeFeeAccumulatorPositionWithLiquidity(ctx sdk.
 	s.Require().NoError(err)
 }
 
+// addLiquidityToUptimeAccumulators adds shares to all uptime accumulators as defined by the `liquidity` parameter.
+// This helper is primarily used to test incentive accrual for specific tick ranges, so we pass in filler values
+// for all other components (e.g. join time and freeze duration).
+func (s *KeeperTestSuite) addLiquidityToUptimeAccumulators(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick, upperTick int64, liquidity []sdk.Dec) {
+	s.Require().Equal(len(liquidity), len(types.SupportedUptimes))
+
+	positionName := string(types.KeyFullPosition(poolId, owner, lowerTick, upperTick, time.Unix(1, 1), 0))
+	uptimeAccums, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulators(ctx, poolId)
+	s.Require().NoError(err)
+
+	for uptimeIndex, uptimeAccum := range uptimeAccums {
+		err := uptimeAccum.NewPosition(positionName, liquidity[uptimeIndex], &accum.Options{})
+		s.Require().NoError(err)
+	}
+}
+
+// addUptimeGrowthInsideRange adds uptime growth inside the range defined by [lowerTick, upperTick).
+//
+// By convention, we add additional growth below the range. This translates to the following logic:
+//
+//   - If currentTick < lowerTick < upperTick, we add to the lower tick's trackers, but not the upper's.
+//
+//   - If lowerTick <= currentTick < upperTick, we add to just the global accumulators.
+//
+//   - If lowerTick < upperTick <= currentTick, we add to the upper tick's trackers, but not the lower's.
+func (s *KeeperTestSuite) addUptimeGrowthInsideRange(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, currentTick, lowerTick, upperTick int64, uptimeGrowthToAdd []sdk.DecCoins) {
+	s.Require().True(lowerTick <= upperTick)
+
+	// Note that we process adds to global accums at the end to ensure that they don't affect the behavior of uninitialized ticks.
+	if currentTick < lowerTick {
+		// Add to lower tick's uptime trackers
+		lowerTickInfo, err := s.App.ConcentratedLiquidityKeeper.GetTickInfo(ctx, poolId, lowerTick)
+		s.Require().NoError(err)
+		s.Require().Equal(len(lowerTickInfo.UptimeTrackers), len(uptimeGrowthToAdd))
+
+		newLowerUptimeTrackerValues, err := addDecCoinsArray(cl.GetUptimeTrackerValues(lowerTickInfo.UptimeTrackers), uptimeGrowthToAdd)
+		s.Require().NoError(err)
+
+		s.initializeTick(ctx, currentTick, lowerTick, lowerTickInfo.LiquidityGross, lowerTickInfo.FeeGrowthOutside, wrapUptimeTrackers(newLowerUptimeTrackerValues), true)
```

Comment:
nit: at first i thought this could be problematic as I thought this assumed the tick was always be initialized and never updated, but realized after stepping into it that this is not the case. We might want to update the name of this helper to reflect this